soft_delete_sample.py

```diff
@@ -215,47 +188,56 @@ def deleted_key_recovery(self):
-        print('created key {}\n{}'.format(key_to_recover, self._serialize(key)))
+        key = key_client.create_key(key_to_recover, 'RSA')
+        print('created key {}'.format(key.name))
 
-        key = self.keyvault_data_client.create_key(vault.properties.vault_uri, key_to_purge, 'RSA')
-        print('created key {}\n{}'.format(key_to_purge, self._serialize(key)))
+        key = key_client.create_key(key_to_purge, 'RSA')
+        print('created key {}'.format(key.name))
 
         # list the vault keys
-        keys = self.keyvault_data_client.get_keys(vault.properties.vault_uri)
-        print('keys: \n{}'.format(self._serialize(keys)))
+        keys = key_client.list_properties_of_keys()
+        print("keys:")
+        for key in keys:
+            print(key.name)
 
         # delete the keys
-        deleted_key = self.keyvault_data_client.delete_key(vault.properties.vault_uri, key_to_recover)
-        self._wait_on_delete_completed(vault.properties.vault_uri, 'key', key_to_recover)
-        print('deleted key {}\n{}'.format(key_to_recover, self._serialize(deleted_key)))
+        key_client.begin_delete_key(key_to_recover).wait()
+        print('deleted key {}'.format(key_to_recover))
 
-        deleted_key = self.keyvault_data_client.delete_key(vault.properties.vault_uri, key_to_purge)
-        self._wait_on_delete_completed(vault.properties.vault_uri, 'key', key_to_purge)
-        print('deleted key {}\n{}'.format(key_to_purge, self._serialize(deleted_key)))
+        key_client.begin_delete_key(key_to_purge).wait()
+        print('deleted key {}'.format(key_to_purge))
 
         # list the deleted keys
-        deleted_keys = self.keyvault_data_client.get_deleted_keys(vault.properties.vault_uri)
-        print('deleted keys: \n{}'.format(self._serialize(deleted_keys)))
+        deleted_keys = key_client.list_deleted_keys()
+        print("deleted keys:")
+        for deleted_key in deleted_keys:
+            print(deleted_key.name)
 
         # recover a deleted key
-        key = self.keyvault_data_client.recover_deleted_key(vault.properties.vault_uri, key_to_recover)
-        self._wait_on_recover_completed(vault.properties.vault_uri, 'key', key_to_recover)
-        print('recovered key {}\n{}'.format(key_to_recover, self._serialize(key)))
+        recover_key_poller = key_client.begin_recover_deleted_key(key_to_recover)
+        recovered_key = recover_key_poller.result()
+        print('recovered key {}'.format(recovered_key.name))
 
         # purge a deleted key
-        self.keyvault_data_client.purge_deleted_key(vault.properties.vault_uri, key_to_purge)
+        key_client.purge_deleted_key(key_to_purge)
+        time.sleep(50)
         print('purged key {}'.format(key_to_purge))
 
         # list the vaults key
-        keys = self.keyvault_data_client.get_keys(vault.properties.vault_uri)
-        print('keys: \n{}'.format(self._serialize(keys)))
+        keys = key_client.list_properties_of_keys()
+        print("keys:")
```

Comment:
To align with print statements from the certificate and secret samples:
```suggestion
        print("all of the keys in the client's vault:")
```